src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -196,21 +200,89 @@ private void addDetailEntriesToNewGroupEntry(PdpFeedGroupEntry newGroupEntry, Pd
             for (PdpFeedAccountingEntry originalAccountingEntry : originalDetailEntry.getAccounting()) {
                 originalDetailTransactionTotal = originalDetailTransactionTotal.add(originalAccountingEntry.getAmount());
-                addOnlyPositiveAccountingEntryForPDPProcessing(newDetailEntry, originalAccountingEntry);
+                newDetailEntry.getAccounting().add(copyAccountingEntry(originalAccountingEntry));
             }
+            cleanAccountingEntriesInDetailEntry(newDetailEntry);
             addNewPaymentDetailToGroupIfTotalIsPositive(newGroupEntry, newDetailEntry, originalDetailTransactionTotal);
         }
     }
-
-    private void addOnlyPositiveAccountingEntryForPDPProcessing(PdpFeedDetailEntry newDetailEntry,
-            PdpFeedAccountingEntry originalAccountingEntry) {
-        if (originalAccountingEntry.getAmount().isPositive()) {
-            newDetailEntry.getAccounting().add(copyAccountingEntry(originalAccountingEntry));
+    
+    protected void cleanAccountingEntriesInDetailEntry(PdpFeedDetailEntry newDetailEntry) {
+        Map<DebitCreditTotal, KualiDecimal> totals = caluclateTotals(newDetailEntry);
+        KualiDecimal originalDebitTotal = totals.get(DebitCreditTotal.DEBIT);
+        KualiDecimal originalCreditTotal = totals.get(DebitCreditTotal.CREDIT);
+        
+        if (areThereMoreCreditsThanDebits(originalDebitTotal, originalCreditTotal)) {
+            LOG.info("cleanAccountingEntriesInDetailEntry, credits are greater than or equal to debits, so removing PDP transactions");
+            newDetailEntry.getAccounting().clear();
+        } else if (shouldDecductCreditsFromDebits(originalDebitTotal, originalCreditTotal)) {
+            LOG.info("cleanAccountingEntriesInDetailEntry, deduct credits from debits required");
+            deductCreditAmountsFromDebitTransactions(newDetailEntry, originalCreditTotal);
         } else {
-            LOG.info("addOnlyPositiveAccountingEntryForPDPProcessing, not adding accounting entry: " + originalAccountingEntry.toString());
+            LOG.info("cleanAccountingEntriesInDetailEntry, no changes required to the accounting entries");
+        }
+        removeNonPositiveAccountingEntriesFromDetailEntry(newDetailEntry);
+    }
+    
+    protected Map<DebitCreditTotal, KualiDecimal> caluclateTotals(PdpFeedDetailEntry detailEntry) {
+        KualiDecimal debitTotal = KualiDecimal.ZERO;
+        KualiDecimal creditTotal = KualiDecimal.ZERO;
+        for (PdpFeedAccountingEntry accountingEntry : detailEntry.getAccounting()) {
+            if (accountingEntry.getAmount().isPositive()) {
+                debitTotal = debitTotal.add(accountingEntry.getAmount());
+            } else {
+                creditTotal = creditTotal.add(accountingEntry.getAmount());
+            }
+        }
+        Map<DebitCreditTotal, KualiDecimal> totals = new HashMap<DebitCreditTotal, KualiDecimal>();
```

Comment:
Since you're using enum constants as the keys, you could alternatively use an EnumMap here.